x/ibc/04-channel/client/utils/utils.go

```diff
@@ -9,45 +9,57 @@ import (
 
 // QueryPacket returns a packet from the store
 func QueryPacket(
-	ctx client.CLIContext, portID, channelID string, sequence, timeout uint64, prove bool,
+	ctx client.CLIContext, portID, channelID string,
+	sequence, timeout uint64, queryRoute string,
 ) (types.PacketResponse, error) {
+	var packetRes types.PacketResponse
+
 	req := abci.RequestQuery{
 		Path:  "store/ibc/key",
 		Data:  types.KeyPacketCommitment(portID, channelID, sequence),
-		Prove: prove,
+		Prove: true,
 	}
 
 	res, err := ctx.QueryABCI(req)
 	if err != nil {
-		return types.PacketResponse{}, err
+		return packetRes, err
 	}
 
-	channelRes, err := QueryChannel(ctx, portID, channelID, prove)
+	channel, err := QueryChannel(ctx, portID, channelID, queryRoute, true)
 	if err != nil {
-		return types.PacketResponse{}, err
+		return packetRes, err
 	}
 
-	destPortID := channelRes.Channel.Counterparty.PortID
-	destChannelID := channelRes.Channel.Counterparty.ChannelID
+	destPortID := channel.Channel.Counterparty.PortID
+	destChannelID := channel.Channel.Counterparty.ChannelID
+
+	var data types.PacketDataI
+	// TODO: commitment data is stored, not the data
+	// but we are unmarshalling the commitment in a json format
+	// because the current ICS 20 implementation uses json commitment form
+	// need to be changed to use external source of packet(e.g. event logs)
```

Comment:
What do event logs have to do with querying? We'll still need to query for the proof, event logs are just to figure out when a relayer *should* query.

Reply:
The relayers need to relay the full packet data so it can be executed on the app level handler on the other side, but the proving is done for the commitment data. The full packet data should be stored somewhere but does not need to be provable.

Reply:
Agreed; same response as above comment.